### PR TITLE
Youtube Atom (née MediaAtom)

### PR DESCRIPTION
**Pull request by nicl**
**UPDATE:** have excluded `Caption` now as the platforms may differ in this and it's a component that is shared more widely too and we'd rather not duplicate it. I've updated the picture accordingly.

## What does this change?

Adds (copied from DC) the Youtube/Media Atom. Note, changes have been made to:

* use Format instead of separate Pillar/Design/Display arguments
* ~~inline icon SVGs (for now)~~ update: use src here where possible

There are a lot of open questions here, which I wanted to get feedback on before continuing:

* ~~do we need *all* of the variants of Caption etc? - I am a bit concerned with the lines of code involved in some of these components, even though they reflect (current) real features/design variations~~
* should we just call this YoutubeAtom? (update: yes)
* how do we handle code reuse for child components, pillar palettes
etc?

## Images

Screenshot 2020-09-07 at 15 23 27

### CODE REVIEW

**Comment by gtrufitt** [comment]
> should we just call this YoutubeAtom?

+1

File: src/MediaAtom/Caption.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import { Format, Pillar, Display } from '@guardian/types/Format';
+
+const TriangleIcon = (
+    <svg width="11" height="10" viewBox="0 0 11 10">
+        <path fillRule="evenodd" d="M5.5 0L11 10H0z" />
+    </svg>
+);
+
+type Props = {
+    format: Format;
```
Comment (gtrufitt) [inline src/MediaAtom/Caption.tsx:18]: Feels weird to have to bring caption into here when it exists in dotcom rendering? I would say that the caption used for all embeds/images/media etc should be the same? +1 on taking a Format!

**Comment by gtrufitt** [comment]
Note on the failing compilation:

`semantic error TS17016: JSX fragment is not supported when using --jsxFactory`

I've seen this elsewhere.

Issue with TS, but can be fixed with babel plugins.
https://github.com/developit/microbundle/pull/623
https://github.com/cision/rover-ui/pull/186

Looks like fixed in TS4.0
https://github.com/microsoft/TypeScript/pull/38720

File: src/MediaAtom/YoutubeOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+const BottomLeft = ({
+    children,
+}: {
+    children: JSX.Element | JSX.Element[];
+}) => (
+    <div
+        className={css`
+            position: absolute;
+            bottom: 12px;
```
Comment (gtrufitt) [inline src/MediaAtom/YoutubeOverlay.tsx:29]: Do these magic numbers relate to any particular thing that you can assign to a var/ comment to give them context?

File: src/MediaAtom/MediaMeta.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { textSans } from '@guardian/src-foundations/typography';
+
+import { pillarPalette } from './pillarPalette';
+import { Pillar } from '@guardian/types/Format';
+
+const Audio = (
```
[the remark this answers is not in the document]
Comment (nicl) [reply src/MediaAtom/MediaMeta.tsx:9]: I think I just copied them to get something working, but I'll switch to the source ones!

File: src/YoutubeAtom/YoutubeAtom.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                iu: `${adTargeting.adUnit || ''}`,
+                cust_params: encodeURIComponent(
+                    constructQuery(adTargeting.customParams),
+                ),
+            },
+        },
+    };
+};
+
+const constructQuery = (query: { [key: string]: any }): string =>
```
Comment (jfsoul) [inline src/YoutubeAtom/YoutubeAtom.tsx:35]: I suppose there isn't anywhere more sensible for this to live?

**Review by jfsoul — APPROVED** [review]
Code looks good. I get the argument for it being called `YoutubeAtom`, but at the same time `Media` is the terminology used elsewhere (i.e. tools) so it might be worth documenting somewhere (where?) that they are intended to be equivalent.

File: src/MediaAtom/YoutubeOverlay.tsx (hunk 2 above)
Comment (nicl) [reply src/MediaAtom/YoutubeOverlay.tsx]: Thanks, I've used `space` here now from Source.

File: src/YoutubeAtom/YoutubeAtom.tsx (hunk 4 above)
Comment (nicl) [reply src/YoutubeAtom/YoutubeAtom.tsx:35]: Yeah, it is pretty generic code. I'm not too concerned about this being duplicated though (if it does end up being useful elsewhere), as it's only a few lines of code. Or else to refactor later as an alternative.

**Pull request merged**